packages/owlbot-bootstrapper/cli/helper.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {CliArgs} from './run-trigger-command';
+import {execSync} from 'child_process';
+
+export class Helper {
+  args: CliArgs;
+  languageContainers: {
+    language: string;
+    languageContainer: string;
+    repoToClone: string;
+  }[];
+
+  constructor(args: CliArgs) {
+    this.args = args;
+    this.languageContainers = [
+      {
+        language: 'nodejs',
+        languageContainer: 'gcr.io/owlbot-bootstrap-prod/node-bootstrapper',
+        repoToClone: 'git@github.com/googleapis/google-cloud-node.git',
+      },
+    ];
+  }
+
+  public async getLanguageSpecificValues() {
+    for (const languageContainer of this.languageContainers) {
+      if (languageContainer.language === this.args['language']) {
+        languageContainer.languageContainer =
+          await this.getFullLatestContainerName(
+            languageContainer.languageContainer
+          );
+        return languageContainer;
+      }
+    }
+
+    // If there was never a match, and there was no language container specified in the CLI, we should throw
+    if (!this.args['languageContainer']) {
+      throw new Error('No language-specific container specified.');
+    }
+
+    return undefined;
+  }
+
+  public async getFullLatestContainerName(container: string) {
+    const latestSha = Buffer.from(
+      execSync(
+        `gcloud container images describe ${container} --format="value(image_summary.digest)"`
```

Comment:
Do we need to shell out to get this data? It'd be preferable to do this in code rather than requiring the runtime to have the `gcloud` CLI installed. Consider using the container registry API client.

Reply:
I didn't see that as [an option](https://cloud.google.com/container-registry/docs/managing). I think it's because container registry is being [deprecated](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr), even though it's the default for pushing up docker images to gcp. I agree that it would be better to not have the gcloud CLI installed but I am not sure there's another way, feel free to point to other documentation if there is any!

Reply:
Ew, I didn't realize that Container Registry only provides a docker API (it would be possible to use a generic docker API client library with added GCP credential headers, but ugh). I have a strong aversion to node libraries shelling out to command line tools and parsing outputs unless absolutely necessary.